### ISSUE TITLE
Correct way to add custom UIMenuItems

### ISSUE THREAD

**Issue by seidju** [badoo/Chatto#434]
Hi, i'm trying to add custom menu items for each cell (like, reply, delete and etc.). But I didn't found any examples of how to correctly do it. Could you please provide some very simple example of this feature?
Thanks a lot.

**Comment by AntonPalich** [comment]
Hey @seidju. Here're the steps:
1. Define a selector for your custom actions. It allows you to access this selector from different places and has a single source of truth:

```swift
struct UIMenuItemCustomActions {
    static let translate = Selector(("translate"))
}
```

2. Add custom action to shared UIMenuController. You can do it in `viewDidLoad` function:

```swift
let translateMenuItem = UIMenuItem(
    title: "Translate",
    action: UIMenuItemCustomActions.translate
)
UIMenuController.shared.menuItems = [translateMenuItem]
```

3. Override `func canPerformMenuControllerAction(_ action: Selector) -> Bool` in a presenter for a message that should display a new menu item.

```swift
override func canPerformMenuControllerAction(_ action: Selector) -> Bool {
    return action == UIMenuItemCustomActions.translate         
}
```

If you want to present a new menu item for text or photo messages then you have to subclass `TextMessagePresenter` or `PhotoMessagePresenter`.

4. Add a function to UICollectionViewCell that should support a new menu item:

```swift
@objc
private func translate() {
}
```

This function will be called when you press a new action in the menu.

I would define a callback in this cell that will be executed when `translate` function is called. It will allow you to provide a callback in a presenter during configuration and handle this action on a presenter level

**Comment by seidju** [comment]
Thanks @AntonPalich , I made a little bit different way, but still thanks a lot!

**Comment by skhavanekar** [comment]
*didn't find

**Comment by dauletkali** [comment]
@seidju , hi! Can you share your code? I tried a lot of ways but still unsuccessful (

**Comment by Denismih** [comment]
Hello, I use chatto 3.5.0 
I made my custom TextMessageMenuItemPresenter  and set it in TextMessagePresenterBuilder. 
I followed the steps 1, 2. I took step 3 in TextMessageMenuItemPresenter.
I use this code for step 4
`extension TextMessageCollectionViewCell {
    @objc func forward(){
        print("Cell forward ext \(textMessageViewModel.text)")
    }
}`
And  I got my custom menu at the end.
But I think extension is not the right place to call my selector. What should I do to perform it in TextMessageMenuItemCustomPresenter : performMenuControllerAction ?
